src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/HtmlHelper.cs

```diff
@@ -522,6 +522,24 @@ public Task RenderPartialAsync(string partialViewName, object model, ViewDataDic
             string templateName,
             object additionalViewData)
         {
+            if (modelExplorer.Metadata.IsEnum && modelExplorer.Model != null)
+            {
+                var value = (modelExplorer.Model as Enum).ToString("d");
+
+                var enumGrouped = modelExplorer.Metadata.EnumGroupedDisplayNamesAndValues;
+                if (enumGrouped != null &&
+                    enumGrouped.Count() > 0)
```

Comment:
Not sure the corner case of an empty `enum` (i.e. one that has no defined values) is worth a condition at all.